### PR TITLE
psp-3910 add logic for activity actions.

### PR DESCRIPTION
**Pull request by devinleighsmith**
re-opening this now that the generic global modal logic has been merged.

### CODE REVIEW

File: backend/dal/Repositories/NoteRepository.cs [hunk 1]
```diff
@@ -95,17 +95,15 @@ public IEnumerable<PimsNote> GetActivityNotes(long entityId)
...
 
-        public void DeleteActivityNotes(long noteId)
+        public void DeleteActivityNotes(long entityId)
         {
-            var activityNotes = this.Context.PimsActivityInstanceNotes.Where(x => x.NoteId == noteId).ToList();
+            var activityNotes = this.Context.PimsActivityInstanceNotes.Where(x => x.ActivityInstanceId == entityId).ToList();
             if (activityNotes.Any())
             {
-                foreach (var note in activityNotes)
+                foreach (var activityNote in activityNotes)
```
Comment (devinleighsmith) [inline backend/dal/Repositories/NoteRepository.cs:103]: @asanchezr please confirm this.
I didn't see requirements for this and if a note still exists in an activity the activity can't be deleted.
If we need notes for auditing is the history table not sufficient?

Comment (asanchezr) [reply backend/dal/Repositories/NoteRepository.cs:103]: I didn't code the delete notes feature. That was @Maheshbabu28 story - he can expand on this decision

Comment (asanchezr) [reply backend/dal/Repositories/NoteRepository.cs:103]: But the presence of **isDisabled** flag seems to indicate that they would want a "soft" delete - or else why would we need it?

Comment (devinleighsmith) [reply backend/dal/Repositories/NoteRepository.cs:103]: I think doug adds it as a default to basically all of our entities.

Comment (devinleighsmith) [reply backend/dal/Repositories/NoteRepository.cs:103]: confirmed, we do not need to soft delete notes. I will remove the logic that sets that flag during deleting in a future story.

File: backend/api/Areas/Activities/Controllers/ActivityController.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        /// </summary>
+        /// <param name="activityId">Used to identify the activity.</param>
+        /// <param name="activityModel">The updated activity values.</param>
+        /// <returns></returns>
+        [HttpPut("{activityId:long}")]
+        [Produces("application/json")]
+        [HasPermission(Permissions.ActivityEdit)]
+        [ProducesResponseType(typeof(ActivityInstanceModel), 200)]
+        [SwaggerOperation(Tags = new[] { "activity" })]
+        public IActionResult UpdateActivity( long activityId, [FromBody] ActivityInstanceModel activityModel)
```
Comment (asanchezr) [inline backend/api/Areas/Activities/Controllers/ActivityController.cs:90]: nit: formatting - extra space after function name `UpdateActivity( long ...)`

File: backend/api/Services/ActivityService.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            ValidateVersion(model.ActivityInstanceId, model.ConcurrencyControlNumber);
+
+            var newActivityInstance = _activityRepository.Update(model);
+            _activityRepository.CommitTransaction();
+            return _activityRepository.GetById(newActivityInstance.ActivityInstanceId);
+        }
+
+        public void Delete(long activityId)
+        {
+            _logger.LogInformation("Deleting activity instance ...", activityId);
```
Comment (asanchezr) [inline backend/api/Services/ActivityService.cs:88]: nit: `"Deleting activity instance {activityId} ..."`

File: frontend/src/features/research/activities/activity/ActivityContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  } = useActivityRepository();
+
+  useDeepCompareEffect(() => {
+    const fetchActivity = async () => {
+      if (!!activityId) {
+        await execute(activityId);
+      }
+    };
+    fetchActivity();
+  }, [activityId, execute]);
```
Comment (asanchezr) [inline frontend/src/features/research/activities/activity/ActivityContainer.tsx:27]: not sure we need `useDeepCompareEffect` here... wouldn't a simple `useEffect` work?

**Review by asanchezr — APPROVED** [review]

**Pull request merged**